core/src/parser/grammar.lalrpop

```diff
@@ -946,9 +946,9 @@ extern {
 
     enum Token<'input> {
         "identifier" => Token::Normal(NormalToken::Identifier(<&'input str>)),
-        "str literal" => Token::Str(StringToken::Literal(<&'input str>)),
+        "str literal" => Token::Str(StringToken::Literal(<String>)),
```

Comment:
Ah, this means that now, we have to allocate each and every string literal, while we could refer to the original input before. It's not necessary a showstopper, but I didn't think about this consequence. Not for this PR, but I wonder if using something like `Cow<'input>` could reduce the memory footprint, if it's ever an issue

Reply:
We are allocating a `String` for each `StrChunk::Literal` in the parser anyway, so this might not be as big of a deal as it seems.

Reply:
But we don't reuse the allocation, do we? I think it's used in the `acc.push_str(&s)` when building a string literal. If I understand correctly, the previous situation was: for each `ChunkLiteralPart` (or whatever it's called) we refer directly to the input, so no allocation, and the when building the actual `ChunkLiteral`, we're accumulating, where we allocate. Now we allocate for each `ChunkLiteralPart`, and then we consume them into `ChunkLiteral`, so the allocation are released but not reused. So we should be doing `total_count_of_chunk_literal_part` new allocations as compared to before? That being said, memory isn't really an issue, because they should be short-lived. It might hurt performance a bit though.

A possiblity would be to make a simple optimization for the case of "only one chunk literal part", where we could directly take the inner String instead of starting from `String::new()` and then pushing it once with `push_str`.